gunrock/app/mf/mf_enactor.cuh

```diff
@@ -7,180 +7,331 @@
+//	      printf("flow[%d] = %lf\n", v, f[v]);
+//	    }, graph.edges, util::DEVICE, oprtr_parameters.stream));
+
+	
+	oprtr_parameters.advance_mode = "ALL_EDGES";
+	GUARD_CU(active.ForAll(
+	    [] __host__ __device__ (SizeT *a, const SizeT &v){
+	      a[v] = 0;
+	    }, 1, util::DEVICE, oprtr_parameters.stream));
+	
+	// ADVANCE_PUSH_OP
+	GUARD_CU(oprtr::Advance<oprtr::OprtrType_V2V>(
+		    graph.csr(), &local_vertices, null_ptr,
+		    oprtr_parameters, advance_push_op));
+	GUARD_CU2(cudaStreamSynchronize(oprtr_parameters.stream),
+	    "cudaStreamSynchronize failed");
+	
+	GUARD_CU(lowest_neighbor.ForAll(
+          [] __host__ __device__ (VertexT *el, const SizeT &v){
+	    el[v] = util::PreDefinedValues<VertexT>::InvalidValue;
+          }, graph.nodes, util::DEVICE, oprtr_parameters.stream));
+	GUARD_CU2(cudaStreamSynchronize(oprtr_parameters.stream),
+          "cudaStreamSynchronize failed");
+
+
+	// ADVANCE_FIND_LOWEST_OP
+	GUARD_CU(oprtr::Advance<oprtr::OprtrType_V2V>(
+		    graph.csr(), &local_vertices, null_ptr,
+		    oprtr_parameters, advance_find_lowest_op)); 
+	GUARD_CU2(cudaStreamSynchronize(oprtr_parameters.stream),
+	    "cudaStreamSynchronize failed");
+      
+//	GUARD_CU(lowest_neighbor.ForAll(
+//          [] __host__ __device__ (VertexT *el, const SizeT &v){
+//            printf("lowest_neighbor[%d] = %d\n", v, el[v]);
+//          }, graph.nodes, util::DEVICE, oprtr_parameters.stream));
+	GUARD_CU2(cudaStreamSynchronize(oprtr_parameters.stream),
+          "cudaStreamSynchronize failed");
+
+	// ADVANCE RELABEL OP
+	GUARD_CU(oprtr::Advance<oprtr::OprtrType_V2V>(
+		    graph.csr(), &local_vertices, null_ptr,
+		    oprtr_parameters, advance_relabel_op));
+	GUARD_CU2(cudaStreamSynchronize(oprtr_parameters.stream),
+	    "cudaStreamSynchronize failed");
+	
+//	GUARD_CU(active.ForAll(
+//	    [] __host__ __device__ (SizeT *a, const SizeT &v){
+//	      if (a[v]){
+//		printf("there are");
+//	      }else{
+//		printf("there are not");
+//	      }
+//	      printf(" active nodes\n");
+//	    }, 1, util::DEVICE, oprtr_parameters.stream));
+
+	GUARD_CU2(cudaStreamSynchronize(oprtr_parameters.stream),
+	    "cudaStreamSynchronize failed");
+
+	//printf("new updated vertices %d\n", frontier.queue_length);
+
+	frontier.queue_reset = true;
+	oprtr_parameters.filter_mode = "BY_PASS";
+	GUARD_CU(oprtr::Filter<oprtr::OprtrType_V2V>(
+		    graph.csr(), frontier.V_Q(), frontier.Next_V_Q(),
+		    oprtr_parameters, 
+	[active]
+	__host__ __device__
+	(const VertexT &src, VertexT &dest, const SizeT &edge_id,
+	  const VertexT &input_item, const SizeT &input_pos,
+	  SizeT &output_pos) -> bool
+	  {
+	    return active[0] > 0;
+	  }));
+
+	frontier.queue_index++;
+	// Get back the resulted frontier length
+	GUARD_CU(frontier.work_progress.GetQueueLength(
+		    frontier.queue_index, frontier.queue_length,
+		    false, oprtr_parameters.stream, true));
+
+	GUARD_CU2(cudaStreamSynchronize(oprtr_parameters.stream),
```

Comment:
I think we need to consider the amount of stream syncs that are happening here, do we need all of these? What can go on without the sync.

Reply:
commented most of them